core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerProcess.java

```diff
@@ -107,10 +106,10 @@
         });
       }
       // In the worst case, each worker factory is blocked waiting for the dependent servers to be
-      // registered at worker registry, so the maximum timeout here is set to the multiply of
-      // the number of factories by the default timeout of getting a worker from the registry.
+      // registered at worker registry. The maximum timeout here is set by configuration
+      // alluxio.worker.startup.timeout
       CommonUtils.invokeAll(callables,
-          (long) callables.size() * 10 * Constants.DEFAULT_REGISTRY_GET_TIMEOUT_MS);
+          (long) callables.size() * ServerConfiguration.getMs(PropertyKey.WORKER_STARTUP_TIMEOUT));
```

Comment:
@beinan Does it make sense to remove the `callables.size() * ` too? There's only one callable currently and that is the BlockWorker. Even if there're more callables, I think it makes more sense to configure the total time instead of doing a multiplication. WDYT?

Reply:
good call, I will post a PR soon.  Thank you @jiacheliu3 !

Reply:
PR https://github.com/Alluxio/alluxio/pull/15062 just posted, @jiacheliu3 could you help take a look?  thanks!